src/hclust.jl

```diff
@@ -590,14 +685,14 @@ function hclust_nn(d::AbstractMatrix, linkage::Function)
         end
         isempty(NN) && push!(NN, 1) # restart NN chain
     end
-    return rorder!(htre.merges)
+    return orderleaves_r!(htre.merges)
```

Comment:
```suggestion
    return htre.merges
```

At the moment all specific hierarchical clustering methods (`hclust_xxx()`) call `rorder!()` at the end.
Since we are introducing alternative ordering options, we should stop doing this and call `orderleaves_r!()` once -- from `hclust()`.